src/main/java/tool/keepertool/ResultPanel.java

```diff
@@ -103,37 +103,34 @@ private void init() {
-                                                                                                                          .getID(),
-                                                                                   core.model.HOVerwaltung.instance()
-                                                                                                                          .getModel()
-                                                                                                                          .getCurrentPlayers(),
-                                                                                   core.model.HOVerwaltung.instance()
-                                                                                                                          .getModel()
-                                                                                                                          .getBasics()
-                                                                                                                          .getDatum());
-                    core.gui.RefreshManager.instance().doReInit();
-                    parent.setVisible(false);
-                    parent.dispose();
-                }
-            });
+        set.addActionListener(arg0 -> {
+            final Player sp = HOVerwaltung.instance().getModel().getCurrentPlayer(id);
+            double decimals = average - sp.getGKskill()
+                              - sp.getSub4Skill(PlayerSkill.KEEPER);
+
+            if (decimals > 1) {
+                decimals = 0.99;
+            }
+
+            if (decimals < 0) {
+                decimals = 0;
+            }
+
+            sp.setSubskill4PlayerSkill(PlayerSkill.KEEPER, (float)decimals);
+            core.db.DBManager.instance().saveSpieler(HOVerwaltung.instance()
+                                                                                                                  .getModel()
+                                                                                                                  .getID(),
+                                                                           HOVerwaltung.instance()
+                                                                                                                  .getModel()
+                                                                                                                  .getCurrentPlayers(),
+                                                                           HOVerwaltung.instance()
+                                                                                                                  .getModel()
+                                                                                                                  .getBasics()
+                                                                                                                  .getDatum().toDbTimestamp());
```

Comment:
Could you reformat this bit please?  It's a bit hard to read.

Reply:
done